src/main/java/org/jenkinsci/plugins/envinject/service/BuildCauseRetriever.java

```diff
@@ -59,16 +61,15 @@ private static void insertRootCauseNames(Set<String> causeNames, Cause cause, in
 
     private static Map<String, String> buildCauseEnvironmentVariables(String envBase, Collection<String> causeNames) {
         Map<String, String> triggerVars = new HashMap<String, String>();
-        StringBuilder all = new StringBuilder();
+        List<String> nonEmptyNames = new ArrayList<String>();
         for (String name : causeNames) {
-            if (!StringUtils.isBlank(name)) {
-                triggerVars.put(envBase + "_" + name, "true");
-                all.append(",");
-                all.append(name);
+            if (isNotBlank(name)) {
+                triggerVars.put(on("_").join(envBase, name), "true");
+                nonEmptyNames.add(name);
             }
         }
         // add variable containing all the trigger names
-        triggerVars.put(envBase, all.toString().substring(1));
+        triggerVars.put(envBase, on(",").join(nonEmptyNames));
```

Comment:
Conflicts with #49 . Please avoid bulk changes (especially when they are not listed in the PRs summary)